src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.h

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+ucc_status_t ucc_tl_probe_ip_over_ib(const char* ib_dev_list, struct
+                                     sockaddr_storage *addr);
+
+#ifndef MCAST_H_
+#define MCAST_H_
+#include "tl_mlx5_mcast_progress.h"
+
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
+static inline int ucc_tl_mlx5_mcast_poll_send(ucc_tl_mlx5_mcast_coll_comm_t *comm)
+{
+    struct ibv_wc wc;
+    int           num_comp;
+    
+    num_comp = ibv_poll_cq(comm->scq, 1, &wc);
+    
+    tl_trace(comm->lib, "polled send completions: %d", num_comp);
+    
+    if (num_comp < 0) {
+
+        tl_error(comm->lib, "send queue poll completion failed %d", num_comp);
+
```

Comment:
return error

Reply:
ucc_tl_mlx5_mcast_poll_send() returns the number of completed send WC. If num_comp is negative, then we return it and the function who is calling it is responsible for error handling